src/game/effects.c

```diff
@@ -143,3 +143,11 @@ void Effect_Draw(int16_t fxnum)
         Matrix_Pop();
     }
 }
+
+void Effect_DoRoutine(int32_t effect_num)
```

Comment:
I'd remove this argument and name the function `Effect_RunActiveFlipEffect` or something. In general I'd say that the API around `g_FlipEffect` is poorly designed and we rely in many places on this singleton, so it's OK to do it here. Refactoring the effects to allow multiple routines to run at the same time would require us to do heavier refactors, and perhaps introduce something in the spirit of `g_Items` – a lot of code for doubtful benefit as I don't think this would ever be leveraged in the OG.